src/Microsoft.VisualStudio.Services.Agent/WebProxy.cs

```diff
@@ -0,0 +1,51 @@
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Common;
+using System;
+using System.IO;
+using System.Net;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    public class WebProxy : IWebProxy
+    {
+        public WebProxy(Uri proxyAddress)
+        {
+            if (proxyAddress == null)
+            {
+                throw new ArgumentNullException(nameof(proxyAddress));
+            }
+
+            ProxyAddress = proxyAddress;
+        }
+
+        public Uri ProxyAddress { get; private set; }
+
+        public ICredentials Credentials { get; set; }
+
+        public Uri GetProxy(Uri destination) => ProxyAddress;
+
+        public bool IsBypassed(Uri uri)
+        { 
+            return false;
+        }
+
+        private static bool _proxySettingsApplied = false;
+
+        public static void ApplyProxySettings()
```

Comment:
as per our talk with @bryanmacfarlane we settled on a config file. we can change it if it becomes a concern when adding auth proxy support.